### PR TITLE
Mrc-2567-hotfixes  add default placeholder to some error report data

### PR DESCRIPTION
**Pull request by LekanAnni**
- Add default value to 
description
step to reproduce and
country

- Make properties of Error report data object mandatory.

- Update tests 

## Type of version change
_Delete as appropriate_

Major - Minor - Patches - None

## Checklist

- [x] I have incremented version number, or version needs no increment
- [x] The build passed successfully, or failed because of ADR tests

### CODE REVIEW

File: src/app/static/src/tests/root/actions.test.ts [hunk 1]
```diff
@@ -213,10 +212,10 @@ describe("root actions", () => {
 
     it("posts error report to teams", async () => {
 
-        const url = "error-report"
+        const url = `error-report`
```
Comment (hillalex) [inline src/app/static/src/tests/root/actions.test.ts:215]: why this change to backticks?

File: src/app/static/src/tests/root/actions.test.ts [hunk 2]
```diff
@@ -238,31 +237,32 @@ describe("root actions", () => {
             })
         });
 
-        const err = mockError("err")
         const getters = {
-            errors: [err]
+            errors: []
```
Comment (hillalex) [inline src/app/static/src/tests/root/actions.test.ts:241]: why remove the error here?

File: src/app/static/src/tests/root/actions.test.ts [hunk 3]
```diff
@@ -319,12 +320,14 @@ describe("root actions", () => {
...
         const expectedError = {error: "OTHER_ERROR", detail: "TestError"};
         expect(commit.mock.calls[0][0]).toEqual({payload: expectedError, type: RootMutation.ErrorReportError});
+        expect(dispatch.mock.calls.length).toBe(0)
         expect(mockAxios.history.post.length).toEqual(1)
         expect(mockAxios.history.post[0].url).toEqual(url)
 
         const expected = {
             email: "some.user@example.com",
             country: "",
+            projectName: "",
```
Comment (hillalex) [inline src/app/static/src/tests/root/actions.test.ts:330]: Since this object is called "expected" this should be "no associated project", country and jobId should be set similarly and then below assert `expect(data.country).toBe(expected.country)` etc.

**Review by hillalex — COMMENTED** [review]
Looks fine, just a bit of tidying up to do in the tests

File: src/app/static/src/tests/root/actions.test.ts [hunk 4]
```diff
@@ -319,14 +321,16 @@ describe("root actions", () => {
...
         const expectedError = {error: "OTHER_ERROR", detail: "TestError"};
         expect(commit.mock.calls[0][0]).toEqual({payload: expectedError, type: RootMutation.ErrorReportError});
+        expect(dispatch.mock.calls.length).toBe(0)
         expect(mockAxios.history.post.length).toEqual(1)
         expect(mockAxios.history.post[0].url).toEqual(url)
 
         const expected = {
             email: "some.user@example.com",
-            country: "",
+            country: "no associated country",
```
Comment (EmmaLRussell) [inline src/app/static/src/tests/root/actions.test.ts:330]: So this test is testing both that the behaviour when the POST fails is as expected, and that the error report defaults are populated? I'd prefer it if that was reflected in the test name or tested in separate tests.

**Review by EmmaLRussell — COMMENTED** [review]
Doesn't look like the code matches the PR description, but it does match what I think needed to happen i.e. default values for country, projectName and jobId (not description and steps to reproduce, which must always be populated from the form)? Also updating `project` to `projectName`?

File: src/app/static/src/tests/root/actions.test.ts (hunk 4 above)
Comment (hillalex) [reply src/app/static/src/tests/root/actions.test.ts:330]: +1 for this being 2 tests please!
* "can return error when error report fails"
* "sends default values when country, project or jobId are missing"

**Review by EmmaLRussell — COMMENTED** [review]
Test failure on buildkite, and it looks like Alex has voted for splitting up that test rather than just a rename (I'd agree that's better approach).

**Review by EmmaLRussell — APPROVED** [review]
Nice. :+1:
I'd say you don't actually need lines 393-419 any more since that's already tested in the new test so the error test can just be that it commits the error correctly - doesn't hurt for it to be there, but would make tests less verbose to remove it.

**Review by hillalex — APPROVED** [review]
